ouster_client/src/client.cpp

```diff
@@ -380,6 +464,55 @@ std::shared_ptr<client> init_client(const std::string& hostname,
     return cli;
 }
 
+std::shared_ptr<client> init_client(const std::string& hostname,                                    
+                                    const std::string& udp_dest_host,
+                                    const std::string& mtp_dest_host,
+                                    lidar_mode ld_mode, timestamp_mode ts_mode,
+                                    int lidar_port, int imu_port,
+                                    int timeout_sec) {
```

Comment:
This works but I do have concerns where a user invokes it from one client which initializes the sensor with certain settings, on another client the user may accidentally change few settings (like ports, ...) which in the very least will interrupt the stream on the first client due to sensor re-init. On the other hand it could probably break the first client if the user selects a different lidar mode for example. I would imaging this would cause a fault on the first client.

We can do reduce the chance of improper use of the API breaking the call into two separate methods (for example `mtp_init_client_main` and `mtp_init_client_slave` or `_secondary`), first method has the option to modify config while the other can't.

On the other hand, we can still use one method but perform due diligence within the provided roslaunch files to protect against improper use. I am leaning towards breaking this API into two separate APIs and cascade the changes to the launch files too so there is no room for misuse. Something like this:

```suggestion
std::shared_ptr<client> mtp_init_client_main(const std::string& hostname,                                    
                                    const sensor_config& config,
                                    const std::string& mtp_dest_host = "",
                                    int timeout_sec) {

std::shared_ptr<client> mtp_init_client_slave(const std::string& hostname,                                    
                                    const std::string& udp_dest_host,
                                    const std::string& mtp_dest_host,
                                    int lidar_port, int imu_port,
                                    int timeout_sec) {
```

I will weigh more on this on Monday.

Reply:
@Samahu check my last commit, please. I tried to split the mtp_init_client api according to your suggestions